app/src/main/java/com/brewthings/app/data/ble/RaptPillParser.kt

```diff
@@ -76,11 +76,18 @@ object RaptPillParser {
             timestamp = Clock.System.now(),
             temperature = (temperature / 128.0 - 273.15).toFloat(),
             gravity = gravity,
-            rawVelocity = if (gravityVelocityValid) gravityVelocity else null,
+            rawVelocity = if (gravityVelocityValid) gravityVelocity.sanitizeVelocity() else null,
             x = x,
             y = y,
             z = z,
             battery = battery,
         )
     }
 }
+
+private fun Float.sanitizeVelocity(): Float? =
+    if (isInfinite() || isNaN() || this > -100 || this < 100) {
```

Comment:
you're checking whether it's invalid so, if the valid range is [-100, 100], you must do
```suggestion
    if (isInfinite() || isNaN() || this < -100 || this > 100) {
```